### PR TITLE
Migrate the Travis configuration to travis-ci.com

### LINKED ISSUE #5041: Update the travis configuration to apply travis-ci.com
We need to update the Travis configuration from travis-ci.org to travis-ci.com.

### PR DESCRIPTION
**Pull request by style95**
## Description
Fix #5041

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#5041)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Review by rabbah — APPROVED** [review]

**Comment by style95** [comment]
I pushed a new commit 3 hours ago, but the CI has not yet started..
It would severely degrade productivity.
If this is because of other jobs running and my job is waiting for it to be finished, we may need to find a way to deal with this.

image

**Comment by rabbah** [comment]
Can we use GitHub actions at all to speed some things up?
I'm surprised old job don't get replaced by new ones - can you cancel them manually?

**Comment by style95** [comment]
@rabbah I reported the issue to the Travis team and they are investigating now.
I tried restarting the build by manual intervention but nothing changed.

**Comment by jiangpengcheng** [comment]
maybe try with github actions is a good option? I did tried with them and it worked well, additionally they are much faster than travis

**Comment by style95** [comment]
Now I only have this test.
image

**Comment by dgrove-oss** [comment]
There's some discussion of GitHub Actions over on builds@apache recently (http://mail-archives.us.apache.org/mod_mbox/www-builds/).  It seems like the backlog to execute GHA is fairly long for repos in the Apache org, so not clear they will be any better than travis.

**Comment by rabbah** [comment]
It's possible to run GHA "on prem" (some other cloud). Would we be able to do that? I can arrange for us to have some compute cycles.

**Comment by style95** [comment]
Tests are failed again due to the rate limit on the Docker hub.

```
4.0: Pulling from library/redis, stderr: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit [marker:invoker_docker.pull_error:52736:52535]
    [2021-01-14T20:05:34.817Z] [ERROR] [#tid_sid_apiMgmt] [StandaloneOpenWhisk] Error starting Api Gatewayorg.apache.openwhisk.core.containerpool.docker.ProcessUnsuccessfulException: info: command was unsuccessful, code: 1 (unsuccessful), stdout: 4.0: Pulling from library/redis
```

IIRC, @rabbah requested an exemption to Docker, and seems it's not yet applied.

Regarding the GHA "on prem", it's essential to have a stable CI environment as it's not easy for contributors to debug an issue with his change if the reason for CI test failures varies every time.

With the indeterministic behavior of Travis CI, I am pro to GHA with the reliable underlying infrastructure.
And as @jiangpengcheng said, Travis CI is generally very slow due to limited resources.

**Comment by style95** [comment]
Finally, it passed all tests.
But I had to rerun some jobs manually and repeatedly to pass tests.

I observed many different reasons:
1. Docker rate limiting
2. CI tests heisenbug
3. Unexpected behavior of Travis's environment.

All of these should be addressed for better productivity.

Also, it relies on python2, we need to migrate to python3 as soon as possible.
Since there is some compatibility issue with python3, @jiangpengcheng `s work https://github.com/apache/openwhisk/pull/5036 is also very important.

File: tools/travis/runStandaloneTests.sh [hunk 1]
```diff
@@ -44,6 +44,8 @@ kind create cluster --wait 5m
 export KUBECONFIG="$(kind get kubeconfig-path)"
 kubectl config set-context --current --namespace=default
 
+# This is required because it is timed out to pull the image during the test.
+docker pull openwhisk/action-nodejs-v10:nightly
```
Comment (style95) [inline tools/travis/runStandaloneTests.sh:48]: We might want to pull all images in advance.

**Pull request merged**